hazelcast-sql/pom.xml

```diff
@@ -91,16 +91,39 @@
                                 </excludes>
                             </artifactSet>
                             <relocations>
+                                <!--
+                                    List relocations for every product explicitly, instead of definining "com" and "org"
+                                    wildcards. Otherwise, the "shade" plugin breaks the generated source code. For example,
+                                    for the wildcard "com." the package "com.google.common" is replaced with the shaded
+                                    pattern twice: "[com].google.[com]mon".
+                                 -->
                                 <relocation>
-                                    <pattern>com.</pattern>
-                                    <shadedPattern>${relocation.root}.com.</shadedPattern>
-                                    <excludes>
-                                        <exclude>com.hazelcast.**</exclude>
-                                    </excludes>
+                                    <pattern>com.fasterxml</pattern>
```

Comment:
IIUC, this relocation now solves them all for https://mvnrepository.com/artifact/com.hazelcast/hazelcast-all/4.1
```

Defect DetectionLGPL |   | com.google.code.findbugs » annotations (optional) | 3.0.1u2 | 3.0.1
-- | -- | -- | -- | --
Cache ClientApache 2.0 |   | javax.cache » cache-api (optional) | 1.1.1 | ✔
LoggingApache 2.0 |   | log4j » log4j (optional) | 1.2.17 | 2.14.0
Apache 2.0 |   | net.openhft » affinity (optional) | 3.2.3 | 3.20.0
LoggingApache 2.0 |   | org.apache.logging.log4j » log4j-api (optional) | 2.13.3 | 2.14.0
LoggingApache 2.0 |   | org.apache.logging.log4j » log4j-core (optional) | 2.13.3 | 2.14.0
JVM LanguagesApache 2.0 |   | org.codehaus.groovy » groovy-all (optional) | 2.1.8 | 3.0.7
JVM LanguagesEPL 2.0GPL 2.0LGPL 2.1 |   | org.jruby » jruby-complete (optional) | 1.7.22 | 9.2.13.0
OSGIApache 2.0 |   | org.osgi » org.osgi.core (optional) | 4.2.0 | 6.0.0
LoggingMIT |   | org.slf4j » slf4j-api (optional) | 1.7.30 | ✔


```
is it correct?

Reply:
Sorry, I didn't get the question, could you please elaborate?

Reply:
Could you check `Compile Dependencies (10)` under this page https://mvnrepository.com/artifact/com.hazelcast/hazelcast-all/4.1? 
It is `0` for https://mvnrepository.com/artifact/com.hazelcast/hazelcast-all/4.0.3 latest release before introducing `sql`. Therefore, we expect them to be `0` as well with these changes. I think it is also verified with this test as well https://github.com/hazelcast/hazelcast/blob/master/hazelcast-all/src/test/java/com/hazelcast/it/DependencyReducedPomAllIT.java, IIUC, this test is also passing in your branch.
hope it clears now.

Reply:
BTW, sorry if I miss some context here, because, I joined as `ad-hoc`.

Reply:
I've just double-checked the produced `pom.xml` file - it has exactly the same dependencies as before the fix. So the PR should be safe to merge.